xwiki-platform-core/xwiki-platform-rest/xwiki-platform-rest-server/src/main/java/org/xwiki/rest/internal/ModelFactory.java

```diff
@@ -977,6 +979,33 @@ private static String serializePropertyValue(PropertyInterface property)
         }
     }
 
+    /**
+     * Serializes the value of the given XObject property. In case the property is an instance of
+     * {@link ComputedFieldClass}, the serialized value is the computed property value.
+     * @param property an XObject property
+     * @param propertyClass the PropertyClass of that XObject proprety
+     * @param context the XWikiContext
+     * @return the String representation of the property value
+     */
+    private static String serializePropertyValue(PropertyInterface property,
+        com.xpn.xwiki.objects.classes.PropertyClass propertyClass, XWikiContext context)
+    {
+        if (propertyClass instanceof ComputedFieldClass) {
+            // TODO: the XWikiDocument needs to be explicitely set in the context, otherwise method
+            // PropertyClass.renderInContext fires a null pointer exception: bug?
+            XWikiDocument document = context.getDoc();
+            context.setDoc(property.getObject().getOwnerDocument());
+            ComputedFieldClass computedFieldClass = (ComputedFieldClass) propertyClass;
+            String value = computedFieldClass.getComputedValue(propertyClass.getName(), "", property.getObject(), context);
+            // Reset the context document to its original value.
```

Comment:
Would be safer to do that in a try/finally.

Reply:
Thanks but I'm not sure what to put in the "finally" clause. In case an exception is raised, should we propagate the exception, or simply return a fake value containing the exception's message? Or don't catch any exception at all? What'd be your suggestion?

Reply:
I did not mentioned `catch` (which is not mandatory). My point is that you should restore the context in a `finally` to be sure you won't leave it with a bad value in case something bad happen.

Reply:
Right, I thought your concern was about the explicit cast to ComputedFieldClass, I had overlooked the context reset aspect. I added a try / finally in the latest commit indeed.